### PR TITLE
Feature/watchlist

### PR DESCRIPTION
**Pull request by scottusrobus**
Adds both watchlist and watching resources (aliases to `customers/:id/watchlist` and `customers/:id/watching` respectively). Example requests:

```javascript
// watching
vhx.watching.items("https://api.vhx.tv/customers/1", function(err, data) {
  // callback
});

// watchlist
vhx.watchlist.items("https://api.vhx.tv/customers/1", function(err, data) {
  // callback
});

vhx.watchlist.addItem({
   customer: "https://api.vhx.tv/customers/1",
   video: "https://api.vhx.tv/videos/1"
}, function(err, data) {
  // callback
});

vhx.watchlist.removeItem({
   customer: "https://api.vhx.tv/customers/1",
   video: "https://api.vhx.tv/videos/1"
}, function(err, data) {
  // callback
});
```

### CODE REVIEW

**Comment by ksheurs** [comment]
👍 !

**Comment by scottusrobus** [comment]
this branch didn't get merged in because there were some outstanding bugs that required some rewriting of the client library. I was timid because we didn't have proper specs setup. But now we do!

screen shot 2017-02-13 at 7 25 46 am

And the watching/watchlist bugs are now fixed. So this should be good to go.

@ksheurs @davegonzalez

File: package.json [hunk 1]
```diff
@@ -22,9 +22,19 @@
...
   },
   "main": "lib/vhx.js",
-  "devDependencies": {},
+  "devDependencies": {
+    "chai": "^3.5.0",
+    "chai-http": "^3.0.0",
+    "dotenv": "^4.0.0",
+    "istanbul": "^0.4.5",
+    "mocha": "^3.2.0",
+    "nyc": "^10.1.2"
```
Comment (ksheurs) [inline package.json]: nyc holla!

**Comment by ksheurs** [comment]
Nice @scottusrobus ! Great to see these specs.

File: test/products.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  //     vhx.products.update(54379, { description: desc }, function(err, updatedProduct) {
+  //       expect(updatedProduct).to.be.a('object');
+  //       expect(updatedProduct).to.be.property('_links');
+  //       expect(updatedProduct).to.be.property('_embedded');
+  //       expect(updatedProduct.description).to.equal(desc);
+  //       expect(updatedProduct._embedded).to.be.property('files');
+  //       done();
+  //     });
+  //   });
+  // });
```
Comment (davegonzalez) [inline test/products.js]: Should this be removed?

Comment (scottusrobus) [reply test/products.js]: yup! deleted

**Comment by davegonzalez** [comment]
😍  looks great!

**Pull request merged**